basic-host-remote/app1/modern.config.js

```diff
@@ -1,28 +1,58 @@
 import appTools, { defineConfig } from '@modern-js/app-tools';
+import { ModuleFederationPlugin } from '@module-federation/enhanced';
+import { RuntimeGlobals } from 'webpack';
 
+const JavascriptModulesPlugin = require('webpack/lib/javascript/JavascriptModulesPlugin');
+
+class AsyncEntryStartupPlugin {
+  apply(compiler) {
+    compiler.hooks.thisCompilation.tap('AsyncEntryStartupPlugin', compilation => {
+      JavascriptModulesPlugin.getCompilationHooks(compilation).renderStartup.tap(
+        'AsyncEntryStartupPlugin',
+        (source, renderContext, upperContext) => {
+          if (upperContext.chunk.hasRuntime()) {
+            return source;
+          }
+          const startup = [
+            'var promiseTrack = [];',
+            `if(__webpack_require__.f && __webpack_require__.f.remotes) __webpack_require__.f.remotes(${JSON.stringify(
+              upperContext.chunk.id,
+            )}, promiseTrack);`,
+            `if(__webpack_require__.f && __webpack_require__.f.consumes) __webpack_require__.f.consumes(${JSON.stringify(
+              upperContext.chunk.id,
+            )}, promiseTrack);`,
+            `var __webpack_exports__ = Promise.all(promiseTrack).then(function() {`,
+            source.source(),
+            'return __webpack_exports__;',
```

Comment:
entrypoints will always return a promise to webpack, if entrypoint is externally required (like commonjs) 

then it will module.exports = new Promise(moduleExports)